sound/soc/amd/acp/acp-sdw-sof-mach.c

```diff
@@ -0,0 +1,1084 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright(c) 2024 Advanced Micro Devices, Inc.
+/*
+ *  amd_sof_sdw - ASOC Machine driver for AMD SoundWire platforms
+ */
+
+#include <linux/bitmap.h>
+#include <linux/device.h>
+#include <linux/dmi.h>
+#include <linux/module.h>
+#include <linux/soundwire/sdw.h>
+#include <linux/soundwire/sdw_type.h>
+#include <sound/soc.h>
+#include <sound/soc-acpi.h>
+#include "amd_sdw_common.h"
+#include "../../sdw_utils/soc_sdw_utils.h"
+#include "../../codecs/rt711.h"
+
+unsigned long sof_sdw_quirk = RT711_JD1;
+static int quirk_override = -1;
+module_param_named(quirk, quirk_override, int, 0444);
+MODULE_PARM_DESC(quirk, "Board-specific quirk override");
+
+static void log_quirks(struct device *dev)
+{
+	if (SOF_JACK_JDSRC(sof_sdw_quirk))
+		dev_dbg(dev, "quirk realtek,jack-detect-source %ld\n",
+			SOF_JACK_JDSRC(sof_sdw_quirk));
+	if (sof_sdw_quirk & SOF_SDW_FOUR_SPK)
+		dev_dbg(dev, "quirk SOF_SDW_FOUR_SPK enabled\n");
+	if (sof_sdw_quirk & SOF_SDW_ACP_DMIC)
+		dev_dbg(dev, "quirk SOF_SDW_ACP_DMIC enabled\n");
+	if (sof_sdw_quirk & SOF_SDW_NO_AGGREGATION)
```

Comment:
also remove

Reply:
we are using SOF_SDW_NO_AGGREGATION flag in multiple places. We are going to change DAI parsing logic . During that clean up we will change the code.